rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs

```diff
@@ -135,3 +141,133 @@ fn test_fetching_blocks() {
+                        .call(CallRequest {
+                            network_identifier: env.network_identifier.clone(),
+                            method_name: "query_block_range".to_owned(),
+                            parameters: ObjectMap::try_from(query_blocks_request.clone()).unwrap(),
+                        })
+                        .await
+                        .unwrap()
+                        .result
+                        .try_into()
+                        .unwrap();
+
+                    assert_eq!(
+                        query_blocks_response.blocks.len(),
+                        std::cmp::min(
+                            MAX_BLOCKS_PER_QUERY_BLOCK_RANGE_REQUEST as usize,
+                            chain_length as usize
+                        )
+                    );
+
+                    // From this point on, we will only test the case where the number of blocks created is greater than 0
+                    if args_with_caller.is_empty() {
+                        return;
+                    }
+                    // Lets try to fetch blocks which are out of scope
+                    query_blocks_request.highest_block_index = highest_block_index * 10;
+                    let query_blocks_response: QueryBlockRangeResponse = env
+                        .rosetta_client
+                        .call(CallRequest {
+                            network_identifier: env.network_identifier.clone(),
+                            method_name: "query_block_range".to_owned(),
+                            parameters: ObjectMap::try_from(query_blocks_request.clone()).unwrap(),
+                        })
+                        .await
+                        .unwrap()
+                        .result
+                        .try_into()
+                        .unwrap();
+
+                    assert_eq!(query_blocks_response.blocks.len() as u64, chain_length);
+
+                    // Lets try to request more blocks than allowed by the max fetchable blocks
+                    query_blocks_request.highest_block_index = highest_block_index;
+                    query_blocks_request.number_of_blocks =
+                        MAX_BLOCKS_PER_QUERY_BLOCK_RANGE_REQUEST + 1;
+                    let query_blocks_response: QueryBlockRangeResponse = env
+                        .rosetta_client
+                        .call(CallRequest {
+                            network_identifier: env.network_identifier.clone(),
+                            method_name: "query_block_range".to_owned(),
+                            parameters: ObjectMap::try_from(query_blocks_request.clone()).unwrap(),
+                        })
+                        .await
+                        .unwrap()
+                        .result
+                        .try_into()
+                        .unwrap();
+
+                    assert_eq!(query_blocks_response.blocks.len(), chain_length as usize);
```

Comment:
Would it also make sense to test with a `highest_block_index` less than the total amount of blocks? And a value of `0` for both the `highest_block_index` and `number_of_blocks`? Perhaps these are already sufficiently tested in the unit/integration tests, in which case feel free to ignore this comment!

Reply:
I'll do that.

Reply:
Done